src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -276,6 +274,22 @@ uct_ib_mlx5_devx_memh_base_address(const uct_ib_mlx5_devx_mem_t *memh)
     return memh->address;
 }
 
+static uint64_t uct_ib_mlx5_devx_memh_iova(const uct_ib_mlx5_devx_mem_t *memh,
+                                           uint32_t iova_offset)
+{
+    uint64_t address;
+
+#if HAVE_IBV_DM
+    if (memh->dm != NULL) {
+        /* Device memory memory key is zero based */
+        return (uint64_t)memh->address + iova_offset;
+    }
+#endif
+
+    address = (uint64_t)uct_ib_mlx5_devx_memh_base_address(memh);
+    return ucs_align_down_pow2(address, UCT_IB_MD_MAX_MR_SIZE) + iova_offset;
```

Comment:
uct_ib_mlx5_devx_reg_ksm_data_contig should adjust the iova it receives according to alignment - instead of doing it here.

Reply:
So we will have to pass `iova_offset` separately to ksm_data_contig

Reply:
instead, maybe we could add the difference between address and ksm_address (i.e how much alignment has moved the address) to iova argument?